### ISSUE TITLE
[Inquiry] Suggestions to improve vTaskDelayUntil

### ISSUE THREAD

**Issue by rmspacefish** [FreeRTOS/FreeRTOS-Kernel#76]
Hello,

I would like to suggest some possible improvements for vTaskDelayUntil.

For our application, we would like to check whether a deadline for a periodic task was missed. 
For us, this is generally the case if vTaskDelayUntil() does not delay the task. Unfortunately, `vTaskDelayUntil` does not have a returnvalue, which tells me whether the task was delayed or not ( could also be implemented by passing a pointer to a boolean), so we needed to perform the check before calling `vTaskDelayUntil` (and we also had to implement the overflow checking again) because the last wake time will be updated internally by it. What do you think about this?

Basically, the interface for `vTaskDelayUntil` would change from

```cpp
void vTaskDelayUntil( TickType_t * const pxPreviousWakeTime, const TickType_t xTimeIncrement)
``` 

to

```cpp
void vTaskDelayUntil( TickType_t * const pxPreviousWakeTime, const TickType_t xTimeIncrement, BaseType_t* xWasDelayed)
``` 

And the local variable `xShouldDelay` is assigned too `*xWasDelayed` at the function end.
Unfortunately, this is an API change but I think it would be a nice feature to implement the checking of missed deadlines.

**Comment by RichardBarry** [comment]
In this case probably best to introduce xTaskDelayUnitl() that proves the additional information - then make vTaskDelayUntil() a macro that calls xTaskDelayUnitl() for backward compatibility.
